app/main.py

```diff
@@ -1,3 +1,55 @@
+from __future__ import annotations
+from typing import Union
+
+
 class Distance:
-    # Write your code here
-    pass
+
+    def __init__(self, km: float = 0.0) -> None:
+        self.km = km
+
+    def __str__(self) -> str:
+        return f"Distance: {self.km} kilometers."
+
+    def __repr__(self) -> str:
+        return f"Distance(km={self.km})"
+
+    @staticmethod
+    def _convert_to_km(other: Union[int, float, Distance]) -> float:
+        if isinstance(other, Distance):
+            return other.km
+        elif isinstance(other, Union[int, float, Distance]):
+            return float(other)
+        raise TypeError("Unsupported type for arithmetic operation")
+
+    def __add__(self, other: Union[int, float, Distance]) -> "Distance":
+        return Distance(self.km + self._convert_to_km(other))
+
+    def __iadd__(self, other: Union[int, float, Distance]) -> "Distance":
+        self.km += self._convert_to_km(other)
+        return self
+
+    def __mul__(self, other: Union[int, float, Distance]) -> "Distance":
+        return Distance(self.km * self._convert_to_km(float(other)))
+
+    def __truediv__(self, other: (int, float)) -> "Distance":
+        if isinstance(other, Distance):
+            raise TypeError("Unsupported type for arithmetic operation")
+        other_km = self._convert_to_km(other)
+        if other_km == 0:
+            raise ZeroDivisionError("Cannot divide by zero")
+        return Distance(round(self.km / other_km, 2))
```

Comment:
I think, you'll get this error anyway in the 40th line if other_km will be 0, so there's no need to raise it